src/main/java/com/selenium/course/pages/AccountDetail.java

```diff
@@ -10,34 +9,17 @@
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
 import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
-import static com.selenium.course.common.Globals.TIMEOUT_MIN;
-import static com.selenium.course.common.Globals.TIMEOUT_NORMAL;
-import com.selenium.course.framework.WebDriverManager;
+import static com.selenium.course.common.Globals.*;
 
-public class AccountDetail {
-	private WebDriver driver;
-	private WebDriverWait wait;
+public class AccountDetail  extends ObjectDetailsPage{
 		
 	@FindBy(xpath = "//h2[contains(.,'Account Detail')]")
     @CacheLookup
-	WebElement accountDetailLabel;
-	
-	@FindBy(id = "acc2_ileinner")
-    @CacheLookup
-	WebElement accountName;
-	
-	@FindBy(name = "delete")
-    @CacheLookup
-	WebElement deleteBtn;
-	
-	@FindBy(name = "edit")
-    @CacheLookup
-	WebElement editBtn;
+	private WebElement accountDetailLabel;
+
 	
-	public AccountDetail() {
-		wait = WebDriverManager.getInstance().getWait();
-		driver = WebDriverManager.getInstance().getDriver();
+	public AccountDetail(WebDriver driver) {
+		super(driver);
 		PageFactory.initElements(driver, this);
```

Comment:
Take advantage from abstract Page class
Move the ```java PageFactory.initElements(driver, this); ``` into Page class in order to inherit it